diff --git a/nova/cmd/manage.py b/nova/cmd/manage.py
index 73d2832c84..cf4be91523 100644
--- a/nova/cmd/manage.py
+++ b/nova/cmd/manage.py
@@ -706,7 +706,7 @@ class CellV2Commands(object):
             # worry about parsing and splitting a URL which could have special
             # characters in the password, which makes parsing a nightmare.
             url = sqla_url.make_url(connection)
-            url.database = url.database + '_cell0'
+            url.set(database='{}_cell0'.format(url.database))
             return urlparse.unquote(str(url))
 
         dbc = database_connection or cell0_default_connection()
